module-apps/application-test/ApplicationTest.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "application-test/ApplicationTest.hpp"
+#include "application-test/TestPopupWindowName.hpp"
+#include "log/log.hpp"
+#include "presenters/TestPresenter.hpp"
+#include "windows/TestPopup.hpp"
+#include "windows/TestWindow.hpp"
+
+namespace app
+{
+    ApplicationTest::ApplicationTest(const std::string &name,
+                                     const std::string &parent,
+                                     StatusIndicators statusIndicators,
+                                     StartInBackground startInBackground)
+        : Application(name, parent, statusIndicators, startInBackground)
+    {
+        LOG_INFO("created!");
+    }
+
+    ApplicationTest::~ApplicationTest()
+    {
+        // this destructor is needed for unique_ptr forward declared TestPresenter
+    }
+
+    sys::ReturnCodes ApplicationTest::InitHandler()
+    {
+        LOG_INFO("initializing!");
+        // NOTE
+        // this is very important! as it:
+        // 1. sets tate initializing on app
+        // 2. initailizes settigsn - every app have settings object by default
+        const auto ret = Application::InitHandler();
+        if (ret != sys::ReturnCodes::Success) {
+            LOG_ERROR("init app failure! %d", int(ret));
+            return ret;
+        }
+        createUserInterface();
+        return sys::ReturnCodes::Success;
+    }
+
+    sys::MessagePointer ApplicationTest::DataReceivedHandler(sys::DataMessage *msgl, sys::ResponseMessage *resp)
+    {
+        // NOTE
+        // If return value here is `nullptr` - then application never responds
+        // othervise it will respond with ResponseMessage
+        auto response = sys::MessagePointer{};
+
+        // NOTE in this place we have all common application uses such as:
+        // * passing key presses to windows
+        // * updading top bar etc
+        response = Application::DataReceivedHandler(msgl);
+        // NOTE
+        // here we would have handleAsyncResponse if this Applicaition would have any `AsyncTask`
+        // if (response && (dynamic_cast<sys::ResponseMessage *>(response.get())->retCode == sys::ReturnCodes::Success))
+        // {
+        //      return response;
+        // }
+        // return handleAsyncResponse(resp);
```

Comment:
just to make sure is it intended to work on `response` which is a retval from Application::DataReceivedHandler(msgl) or should we work on `resp` which is an argument here? I do not understand that comment+part of code together

Reply:
Thanks, resp/response - these just are return values names - I'll fix them. 
What's important is that - no handling of task async response will cause Application AsyncTask useless with no real information why it doesn't work.

In general - DataReceivedHandler is mayor cause of apps windows not responding when you think these should :)